### ISSUE TITLE
Can we draw the polygon borders?

### ISSUE THREAD

**Issue by SimonCoulombe** [r-spatial/leafgl#3]
I've seen this issue on the original leaflet.glify, but I'm not sure how to implement it:
https://github.com/robertleeplummerjr/Leaflet.glify/issues/14

cheers!

**Comment by tim-salabim** [comment]
I am not sure either. I think it would be best if you add your request to the issue you mention above. It would surely be nice to have polygon borders.

**Comment by SimonCoulombe** [comment]
will do!
Oh.  and I forgot to thank you for your work on this package.  I only found it yesterday and already made my first 60k polygons map.  I'm hoping for  1 million now (.canadian postal codes), but I don't want to get my hope to high :)

**Comment by tim-salabim** [comment]
I think you saw my post about 1.8 mio building footprints on twitter? Depending on polygon comexity you might need to chunk the data. There's also a faster way using data.table described in the inst/experiments folder of this repo.

I am very interested to hear back about any experience with this package so don't be shy to post your experience here.

**Comment by tim-salabim** [comment]
@SimonCoulombe did you manage to plot the canadian zip codes?

**Comment by SimonCoulombe** [comment]
Hi Tim,
Thanks for following up!  I had to steer my time toward other projects, so mapping the postal codes has been on the back burner for a bit.    I'll try to get to it when we get back from the holidays.

Merry Christmas (or holidays, whichever suits you better) :)

**Comment by tim-salabim** [comment]
The upstream repo has a PR that has stalled a little, but maybe this will be added at some stage.

https://github.com/robertleeplummerjr/Leaflet.glify/pull/32

**Comment by nevilamos** [comment]
Hi Tim,
stillfairly new to github, trying to follow the outcome of issue#3  drawing borders - I would like to be able to draw just polygon borders with no fill.  is this possible?
thanks

**Comment by tim-salabim** [comment]
Yes! You just need to `st_cast` your polygons to linestrings and then use `addGlPolylines`

**Comment by nevilamos** [comment]
Hi thanks for the quick response,

that works as a sort of workaround, however drawing the linestrings will
result in different behavour to the leflet drawing polygons with a boundary
- particularly with click on or hover over  polygon attributes?

Would be wonderful if it could work in the same way ti does in addPolygons
with ability to have separate colour for fill and line for a polygon.

Thanks for all your great work on this project anyway - At least I can
render all my polygons!

**Comment by nevilamos** [comment]
I got this how I wanted eventually by drawing the ploygons  completely transparent ( so I could still clikc on them to get attributes and also drawing the lines.  Would be nice to roll into one as in addPolygons though.

**Comment by tim-salabim** [comment]
When you use mapview with `mapviewOptions(platform = "leafgl")` you get both the polygons and the borders. I don't want to make this the default in leafgl, because a fix should not be too far away upstream. I think for making maps "manually" using leaflet et al. it is acceptable to do that one step manually. 

I'll leave this open until we have a baked in solution for the polygon borders

**Comment by dholstius** [comment]
+1 for having borders baked in. HTML size is doubled when adding the polyline borders separately, "on top" of the polygons, via a subsequent call to `addGlPolylines()`. Guessing this is because all of the geometry is duplicated!

**Comment by tim-salabim** [comment]
@dholstius yes the reason for doubling the size is that everything, especially the geometry data, is included twice. If I'm not mistaken, there is already a PR in the upstream repo to allow for drawing borders without having to add the data twice. So the current implementation here is a temporal workaround until we get that functionality cascaded down from upstream Leaflet.glify.
